### DISCUSSION TITLE
Is there a better way to get Request body on the "onAfterResponse"?

### DISCUSSION THREAD

**Discussion by kh411d** [go-resty/resty#566] [Q&A] [answered]
Hi, a question why can't I use request body after the response for logging,

This is what I currently implement,

```
func getClient() *resty.Client {
  
  var requestBody interface{}

  return resty.New().EnableTrace().
                         OnBeforeRequest(func(c *resty.Client, r *resty.Request) error {
				if r.Body != nil {
					requestBody = r.Body
				}
				return nil 
			}).OnAfterResponse(func(c *resty.Client, response *resty.Response) error {
                           log(requestBody, response.Body())
                           ...
                            return nil 
			})
}
```

Hence the `var requestBody interface{}` since the `response.Request.Body` always empty

**UPDATE**

Oh wait, I think the `RawRequest.GetBody()` from the `OnAfterResponse` already has this type `io.nopCloser`
```
OnAfterResponse(func(c *resty.Client, r *resty.Response) error {
				req := r.Request
				rawReqBody, _ := req.RawRequest.GetBody() // io.nopCloser

```
I think the Resty library got updated, I suppose the request body is already copied, isn't it?
https://github.com/go-resty/resty/blob/7dda6159eceee10c6cacc6f9d1637042f67423b0/middleware.go#L521-L543

**Comment by kecci** [comment, marked as answer]
Yes you right on based on your new updates, it caused body used if you called the body, 
So you have to put it back to the request body again.
